[linux] wayland build issues

**Describe the bug**

Multiple headers have the `types` declaration.

```
In file included from ../../Kinc/Backends/System/Linux/Sources/kinc/backend/linuxunit.c:14:
../wayland/wayland-generated/wayland-tablet.c.h:53:35: error: redefinition of ‘types’
 static const struct wl_interface *types[] = {
                                   ^~~~~
In file included from ../../Kinc/Backends/System/Linux/Sources/kinc/backend/linuxunit.c:13:
../wayland/wayland-generated/wayland-relative-pointer.c.h:44:35: note: previous definition of ‘types’ was here
 static const struct wl_interface *types[] = {
                                   ^~~~~
In file included from ../../Kinc/Backends/System/Linux/Sources/kinc/backend/linuxunit.c:15:
../wayland/wayland-generated/wayland-viewporter.c.h:43:35: error: redefinition of ‘types’
 static const struct wl_interface *types[] = {
                                   ^~~~~
In file included from ../../Kinc/Backends/System/Linux/Sources/kinc/backend/linuxunit.c:14:
../wayland/wayland-generated/wayland-tablet.c.h:53:35: note: previous definition of ‘types’ was here
 static const struct wl_interface *types[] = {
                                   ^~~~~
In file included from ../../Kinc/Backends/System/Linux/Sources/kinc/backend/linuxunit.c:17:
../wayland/wayland-generated/xdg-decoration.c.h:43:35: error: redefinition of ‘types’
 static const struct wl_interface *types[] = {
                                   ^~~~~
In file included from ../../Kinc/Backends/System/Linux/Sources/kinc/backend/linuxunit.c:15:
../wayland/wayland-generated/wayland-viewporter.c.h:43:35: note: previous definition of ‘types’ was here
 static const struct wl_interface *types[] = {
                                   ^~~~~
In file included from ../../Kinc/Backends/System/Linux/Sources/kinc/backend/linuxunit.c:18:
../wayland/wayland-generated/xdg-shell.c.h:53:35: error: redefinition of ‘types’
 static const struct wl_interface *types[] = {
                                   ^~~~~
In file included from ../../Kinc/Backends/System/Linux/Sources/kinc/backend/linuxunit.c:17:
../wayland/wayland-generated/xdg-decoration.c.h:43:35: note: previous definition of ‘types’ was here
 static const struct wl_interface *types[] = {
```

**Execution Environment:**
- Host system (where you compile your code): debian 9 system
- Kinc revision: e72a68982236f10c90babab893b47f8997c92784

**Additional context**

According to Zeta, it's the wayland-scanner binary that's too old. I'm running `wayland-scanner 1.16.0` that seems to be the latest for debian 9.

--- a/Backends/System/Linux/Sources/kinc/backend/linuxunit.c
+++ b/Backends/System/Linux/Sources/kinc/backend/linuxunit.c
@@ -3,19 +3,11 @@
 #endif
 #include "funcs.h"
 #include <dlfcn.h>
-// #include <kinc/log.h>
 
 #ifndef KINC_NO_WAYLAND
 #include "wayland/display.c.h"
 #include "wayland/system.c.h"
-#include <wayland-generated/wayland-pointer-constraint.c.h>
-#include <wayland-generated/wayland-protocol.c.h>
-#include <wayland-generated/wayland-relative-pointer.c.h>
-#include <wayland-generated/wayland-tablet.c.h>
-#include <wayland-generated/wayland-viewporter.c.h>
 #include "wayland/window.c.h"
-#include <wayland-generated/xdg-decoration.c.h>
-#include <wayland-generated/xdg-shell.c.h>
 #endif
 
 #ifndef KINC_NO_X11

--- a/kfile.js
+++ b/kfile.js
@@ -341,7 +341,7 @@ else if (platform === Platform.Linux || platform === Platform.FreeBSD) {
 		function wl_protocol(protocol, file) {
 			const backend_path = path.resolve(waylandDir);
 			const protocol_path = path.resolve("/usr/share/wayland-protocols", protocol);
-			child_process.spawn("wayland-scanner", ["private-code", protocol_path, path.resolve(backend_path, file + ".c.h")]).on("exit", code => {
+			child_process.spawn("wayland-scanner", ["private-code", protocol_path, path.resolve(backend_path, file + ".c")]).on("exit", code => {
 				if (code != 0) {
 					console.warn("Failed to generate wayland protocol files for", protocol);
 				}
@@ -353,7 +353,7 @@ else if (platform === Platform.Linux || platform === Platform.FreeBSD) {
 			});
 		}
 
-		child_process.spawn("wayland-scanner", ["private-code", "/usr/share/wayland/wayland.xml", path.resolve(waylandDir, "wayland-protocol.c.h")]);
+		child_process.spawn("wayland-scanner", ["private-code", "/usr/share/wayland/wayland.xml", path.resolve(waylandDir, "wayland-protocol.c")]);
 		child_process.spawn("wayland-scanner", ["client-header", "/usr/share/wayland/wayland.xml", path.resolve(waylandDir, "wayland-protocol.h")]);
 		wl_protocol("stable/viewporter/viewporter.xml", "wayland-viewporter");
 		wl_protocol("stable/xdg-shell/xdg-shell.xml", "xdg-shell");
@@ -363,6 +363,7 @@ else if (platform === Platform.Linux || platform === Platform.FreeBSD) {
 		wl_protocol("unstable/relative-pointer/relative-pointer-unstable-v1.xml", "wayland-relative-pointer");
 
 		project.addIncludeDir(path.join(targetDirectory, 'wayland'));
+		project.addFile(waylandDir + "/**");
 	}
 	else if (platform === Platform.FreeBSD) {
 		addBackend('System/FreeBSD');
@@ -370,7 +371,7 @@ else if (platform === Platform.Linux || platform === Platform.FreeBSD) {
 		project.addExclude('Backends/System/Linux/Sources/kinc/backend/input/gamepad.h');
 		project.addDefine("KINC_NO_WAYLAND");
 	}
-	
+
 	if (graphics === GraphicsApi.Vulkan) {
 		g4 = true;
 		g5 = true;

